### PR TITLE
New package: nfs4-acl-tools-0.3.5

### PR DESCRIPTION
**Pull request by gspe**

### CODE REVIEW

File: srcpkgs/nfs4-acl-tools/template [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+pkgname=nfs4-acl-tools
+version=0.3.5
+revision=1
+build_style=configure
+configure_args="--prefix=/usr"
+hostmakedepends="automake autoconf"
+makedepends="libtool attr-devel"
+short_desc="Client tools for manipulating NFSv4 ACLs directly"
+maintainer="Giuseppe Fierro <gspe@ae-design.ws>"
+license=" BSD-3-Clause, LGPL-2.1, GPL-2.0-or-later"
```
Comment (maxice8) [inline srcpkgs/nfs4-acl-tools/template]: LGPL-2.1 isn't a SPDX short-identifier

File: srcpkgs/nfs4-acl-tools/template [hunk 2]
```diff
@@ -0,0 +1,18 @@
+# Template file for 'nfs4-acl-tools'
+pkgname=nfs4-acl-tools
+version=0.3.5
+revision=1
+build_style=configure
+configure_args="--prefix=/usr"
+hostmakedepends="automake autoconf"
+makedepends="libtool attr-devel"
```
Comment (maxice8) [inline srcpkgs/nfs4-acl-tools/template]: libtool shouldn't be in makedepends it only has binaries

File: srcpkgs/nfs4-acl-tools/template (hunk 1 above)
Comment (gspe) [reply srcpkgs/nfs4-acl-tools/template]: Thanks, I forgot to change this.
I have a question, how do you see if a license is *-or-later or *-only version?

**Comment by maxice8** [comment]
On Wed, Sep 19, 2018 at 06:34:29AM -0700, Giuseppe Fierro wrote:
> gspe commented on this pull request.
> 
> 
> 
> > @@ -0,0 +1,18 @@
> +# Template file for 'nfs4-acl-tools'
> +pkgname=nfs4-acl-tools
> +version=0.3.5
> +revision=1
> +build_style=configure
> +configure_args="--prefix=/usr"
> +hostmakedepends="automake autoconf"
> +makedepends="libtool attr-devel"
> +short_desc="Client tools for manipulating NFSv4 ACLs directly"
> +maintainer="Giuseppe Fierro <gspe@ae-design.ws>"
> +license=" BSD-3-Clause, LGPL-2.1, GPL-2.0-or-later"
> 
> Thanks, I forgot to change this.
> I have a question, how do you see if a license is *-or-later or *-only version?
> 
> -- 
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub:
> https://github.com/void-linux/void-packages/pull/2848#discussion_r218804811

See 'Standard License Header' at the bottom of the page of these 2 pages, one for -or-later other for -only

-only -> https://spdx.org/licenses/AGPL-3.0-only.html
-or-later -> https://spdx.org/licenses/AGPL-3.0-or-later.html

the -or-later will have 

or (at your option) any later version. 

those Standard License Headers are put in the source code files of the application.

long story short check your source code files for those headers

**Comment by gspe** [comment]
Ok, thanks.
Can you also help me with this arm platform error?

File: srcpkgs/nfs4-acl-tools/template (hunk 1 above)
Comment (Johnnynator) [reply srcpkgs/nfs4-acl-tools/template]: You have to take a look at the source files e.g. http://git.linux-nfs.org/?p=bfields/nfs4-acl-tools.git;a=blob;f=include/buildrules;h=6dfacde501cc88578811781ff94da629671662db;hb=HEAD is GPL-2.0-only. I cannot find any file that is LGPL licensed.

Comment (gspe) [reply srcpkgs/nfs4-acl-tools/template]: @Johnnynator thanks.
Other licenses are mentioned in COPYING file:
http://git.linux-nfs.org/?p=bfields/nfs4-acl-tools.git;a=blob;f=COPYING;h=3b4e71a9d724c3500c41027c5ba2877de98baf76;hb=f83c544cc6d777de8dbdde2a0be76cf5895aae25

Comment (Johnnynator) [reply srcpkgs/nfs4-acl-tools/template]: @gspe I know. The LGPL part is probably just copied from acl. The only important part is the BSD License since only the build system is GPL which is not in the final package. (Also acl is LGPL-2.1-or-later)

**Comment by newbluemoon** [comment]
The problem seems to be that the build style is actually `gnu-configure`, but `configure` is invoked in the makefile! So first patch `Makefile` to replace the hard-coded options with a place holder for Void’s:

```
--- Makefile.orig
+++ Makefile
@@ -61,16 +61,7 @@
 
 $(CONFIGURE): aclocal.m4
 	autoconf
-	./configure \
-		--prefix=/ \
-		--exec-prefix=/ \
-		--sbindir=/bin \
-		--bindir=/usr/bin \
-		--libdir=/lib \
-		--libexecdir=/usr/lib \
-		--includedir=/usr/include \
-		--mandir=/usr/share/man \
-		--datadir=/usr/share \
+	./configure @configure_args@ \
 		$$LOCAL_CONFIGURE_OPTIONS
 	touch .census
```

Then in the template you can replace `@configure_args@` with Void’s target specific ones:

```
do_configure() {
	sed -i "s;@configure_args@;$(echo $configure_args | paste);" Makefile
}
```

Note: `sed` didn’t like the multi-line `$configure_args` so this construct converts it to a single line.

Change `build_style` to `gnu-configure` and remove `--prefix=/usr` from `configure_args` (it’s set by the build-style now). Use these instead:
```
configure_args="ac_cv_path_LIBTOOL='/usr/bin/libtool --tag=CC'"
 ac_cv_func_malloc_0_nonnull=yes"
```

`libtool` needs to have the `--tag` option set when cross compiling. And `configure` cannot check for `malloc` when cross compiling (sets it to ‘no’ per default), but from a native build we know it’s there so override the test.

Then it should work. :)

**Comment by gspe** [comment]
@newbluemoon I've applied your patch and locally compile fine but it doesn't compile anymore on the builder.

**Comment by newbluemoon** [comment]
Just re-tested locally. It seems that `disable_parallel_build=yes` is necessary, because with more than one make jobs it sometimes works, sometimes not.

File: srcpkgs/nfs4-acl-tools/template [hunk 3]
```diff
@@ -0,0 +1,24 @@
+# Template file for 'nfs4-acl-tools'
+pkgname=nfs4-acl-tools
+version=0.3.5
+revision=1
+build_style=gnu-configure
+configure_args="ac_cv_path_LIBTOOL='/usr/bin/libtool --tag=CC'
+				ac_cv_func_malloc_0_nonnull=yes"
```
Comment (newbluemoon) [inline srcpkgs/nfs4-acl-tools/template]: Should be indented with only a single space.

**Comment by gspe** [comment]
@newbluemoon @maxice8 , now should be ok

File: srcpkgs/nfs4-acl-tools/template [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+makedepends="attr-devel"
+short_desc="Client tools for manipulating NFSv4 ACLs directly"
+maintainer="Giuseppe Fierro <gspe@ae-design.ws>"
+license=" BSD-3-Clause, LGPL-2.1-or-later, GPL-2.0-only"
+homepage="http://linux-nfs.org/wiki/index.php/Main_Page"
+distfiles="http://linux-nfs.org/~bfields/${pkgname}/${pkgname}-${version}.tar.gz"
+checksum=b54a1dbaeb0fd480f2b0d2767b789fa80110124b71c6a73905ade99e9229a47e
+
+do_configure() {
+ sed -i "s;@configure_args@;$(echo $configure_args | paste);" Makefile
```
Comment (maxice8) [inline srcpkgs/nfs4-acl-tools/template]: indent here is wrong

File: srcpkgs/nfs4-acl-tools/template [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+}
+
+post_install() {
+ vlicense COPYING
```
Comment (maxice8) [inline srcpkgs/nfs4-acl-tools/template]: and here as well, it is 1 tab

**Comment by gspe** [comment]
@maxice8 Now should be ok

File: srcpkgs/nfs4-acl-tools/template [hunk 6]
```diff
@@ -0,0 +1,24 @@
+# Template file for 'nfs4-acl-tools'
+pkgname=nfs4-acl-tools
+version=0.3.5
+revision=1
+build_style=gnu-configure
+configure_args="ac_cv_path_LIBTOOL='/usr/bin/libtool --tag=CC'
+	ac_cv_func_malloc_0_nonnull=yes"
```
Comment (maxice8) [inline srcpkgs/nfs4-acl-tools/template]: one whitespace identation

File: srcpkgs/nfs4-acl-tools/template [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+disable_parallel_build=yes
+hostmakedepends="automake autoconf libtool"
+makedepends="attr-devel"
+short_desc="Client tools for manipulating NFSv4 ACLs directly"
+maintainer="Giuseppe Fierro <gspe@ae-design.ws>"
+license=" BSD-3-Clause, LGPL-2.1-or-later, GPL-2.0-only"
```
Comment (maxice8) [inline srcpkgs/nfs4-acl-tools/template]: remove extra whitespace
